### ISSUE TITLE
Replace null actorrefs with ActorRefs.Nobody by default?

### ISSUE THREAD

**Issue by rogeralsing** [akkadotnet/akka.net#989]
The API is somewhat unclear in some areas.
e.g. if explicitly passing null to `Tell` an exception is thrown, which seems OK.

But consider this:

```csharp
//in some message handler:
Context.System.Scheduler.ScheduleTellOnce(SomeTime, receiver, someMessage, null);
```

if the user pass `null` as the sender, as a new developer probably would if they dont want to pass a sender, the scheduler will after `SomeTime` invoke `receiver.Tell(someMessage,null)` which throws, causing an invisible exception in the scheduler.

Should we treat `null` as `ActorRefs.Nobody` by default?

btw. @HCanber has a comment on this in the scheduler code 
`        //TODO: Should we log other exceptions? /@hcanber`

(Ok there might be two different things being discussed there, logging in scheduler and/or replace null with nobody)

**Comment by stefansedich** [comment]
Roger,

I was bit by this the other day, I was meaning to post about it but got
completely swamped with my day to day, I agree that we should probably
handle this in case someone passes null through instead of throwing would
make the API less prickly.


Cheers
Stefan

On Tue, May 19, 2015 at 6:45 PM, Roger Johansson <notifications@github.com>
wrote:

> The API is somewhat unclear in some areas.
> e.g. if explicitly passing null to Tell an exception is thrown, which
> seems OK.
>
> But consider this:
>
> //in some message handler:
> Context.System.Scheduler.ScheduleTellOnce(SomeTime, receiver, someMessage, null);
>
> if the user pass null as the sender, as a new developer probably would if
> they dont want to pass a sender, the scheduler will after SomeTime invoke
> receiver.Tell(someMessage,null) which throws, causing an invisible
> exception in the scheduler.
>
> Should we treat null as ActorRefs.Nobody by default?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akkadotnet/akka.net/issues/989>.
>



-- 
Cheers
Stefan

**Comment by Aaronontheweb** [comment]
> Should we treat null as ActorRefs.Nobody by default?

Yep, I think this is a great idea.

**Comment by stefansedich** [comment]
This should have been fixed by #993 closing it off.
